pkg/addon/addon.go

```diff
@@ -1072,6 +1092,210 @@ func (h *Installer) installDependency(addon *InstallPackage) error {
 	return nil
 }
 
+// validateAddonDependencies checks if addon's dependencies can be satisfied.
+// If dependency is installed, check if the version matches required version.
+// If dependency is not installed, check available addons for dependency
+// matching the required version.
+// Return error if any dependency cannot be satisfied.
+func validateAddonDependencies(addon *InstallPackage, installedAddons addonInfoMap, availableAddons addonInfoMap) error {
+	dependenciesInstalled := []Dependency{}
+	dependenciesNotInstalled := []Dependency{}
+	for _, dep := range addon.Dependencies {
+		_, ok := installedAddons[dep.Name]
+		if ok {
+			dependenciesInstalled = append(dependenciesInstalled, *dep)
+		} else {
+			dependenciesNotInstalled = append(dependenciesNotInstalled, *dep)
+		}
+	}
+
+	// check dependencies against installed addon versions
+	// if dependency is installed, check if the version matches required version
+	var err error
+	for _, dep := range dependenciesInstalled {
+		// get installed version
+		installedAddon := installedAddons[dep.Name]
+		// versions length must be 1
+		if len(installedAddon.AvailableVersions) != 1 {
+			err = multierr.Append(err, errors.New("installedAddon.Versions length must be 1"))
+			continue
+		}
+		installedVersion := installedAddon.AvailableVersions[0]
+
+		match, _ := checkSemVer(installedVersion, dep.Version)
```

Comment:
For compatibility, some existed addons may not specify `dep.Version`. What  `match` value will be returned? 
For those existed addons without 'dep.Version`, do we need reinstall their dependencies?

Reply:
The expected behavior is for `dep.Version == nil` to accept any version. The function currently works this way, and there are test cases to cover this.

Currently this relies on `checkSemVer` returning `true` when 2nd param (`required`) is nil. However, I can make this more explicit by adding this check before calling `checkSemVer`.

Reply:
I updated this function to continue (skip version check) when `dep.Version == nil`